### PR TITLE
feat: add onSelect, show focus of grid rows

### PR DESCRIPTION
**Pull request by chiefcll**
## Description

Add a new onSelect which is used for a Signal to set the selectedIndex or the Row/Column on focus and when it key handling changes this.selected. Useful for Row to be "active" when combined with createSelector as shown in story.

Combined Row & Column types into single Grid type - then the functions can get a GridElement and user has access to the additional props. Keeping them separate made it harder to have a single handleNavigation as the types being passed into the callbacks would have been different. And they are identical types...

### CODE REVIEW

**Comment by chiefcll** [comment]
@anthony9187 - can you finish this up with some prettier colors for the row highlight 👍

**Comment by chiefcll** [comment]
@anthony9187 - This should all be working. LMK your thoughts. I have an idea to just export ElementNode in the future rather than having a duplicate interface which causes headaches. I also noticed there is another issue with the paths when we build...
`types/interfaces.js` isn't found - TS doesn't complain but when we build the TS definitions aren't found. I'll have another PR to fix that

File: packages/solid/components/Column/Grid.types.ts [hunk 1]
```diff
@@ -31,15 +32,29 @@ export interface ColumnProps extends UIComponentProps, ColumnStyleProperties {
   /** function to be called on up click */
   onUp?: KeyHandler;
 
+  /** function to be called on down click */
+  onLeft?: KeyHandler;
+
+  /** function to be called on up click */
+  onRight?: KeyHandler;
+
```
Comment (cee-money) [inline packages/solid/components/Column/Grid.types.ts:40]: extremely minor update

```suggestion
  /** function to be called on left click */
  onLeft?: KeyHandler;

  /** function to be called on right click */
  onRight?: KeyHandler;

```

File: packages/solid/recipes/Grid.stories.tsx [hunk 2]
```diff
@@ -74,3 +75,26 @@ export const Example: StoryObj = {
...
+  render: () => {
+    const Colors = [0xa32eacff, 0x23adccff, 0x23cc37ff, 0xcd8518ff];
+    const highlightRow = { color: 0x00ff00ff, active: { color: 0xff0000ff } };
+    const [selectedIndex, setSelectedIndex] = createSignal(0);
+    const isSelected = createSelector(selectedIndex);
+    return (
+      <Column autofocus plinko onSelect={setSelectedIndex}>
+        <For each={Colors}>
+          {(color, index) => (
+            <Row height={100} states={{ active: isSelected(index()) }} style={highlightRow}>
```
Comment (cee-money) [inline packages/solid/recipes/Grid.stories.tsx:89]: do we want to limit the highlight row style to the active row to show the state change? looks like it's on all rows atm

Screenshot 2024-09-23 at 11 11 08 AM

Comment (chiefcll) [reply packages/solid/recipes/Grid.stories.tsx:89]: I'll double check this... Should be on just the active row.

Comment (chiefcll) [reply packages/solid/recipes/Grid.stories.tsx:89]: Fixed

image

**Review by anthony9187 — APPROVED** [review]
the changes added here should also resolve the typescript issues we we're seeing

**Review by cee-money — APPROVED** [review]

**Pull request merged**